source/opt/dominator_analysis.cpp

```diff
@@ -55,11 +55,30 @@ bool DominatorAnalysisBase::Dominates(Instruction* a, Instruction* b) const {
     return tree_.Dominates(bb_a, bb_b);
   }
 
-  Instruction* current_inst = a;
-  while ((current_inst = current_inst->NextNode())) {
-    if (current_inst == b) {
+  // We handle OpLabel instructions explicitly since they are not stored in the
+  // instruction list.
+  if (tree_.IsPostDominator()) {
```

Comment:
So it looks like post-dominators were broken for instructions in the same block.

This can be simplified to:
```
Instruction *current_inst = a;
Instruction *other_inst = b;
if (tree_.IsPostDominator())
  std::swap(current_inst, other_inst);

while ((current_inst = current_inst->NextNode()))) {
  if (current_inst == other_inst)
    return true;
}

return false;
```

Reply:
Indeed, fixed in a076b9d.